k8s-daemonset/k8s/servicemesh.yml

```diff
@@ -0,0 +1,314 @@
+# expected to send traffic to their node local Linkerd: $NODE_NAME:4140 for
+# HTTP, $NODE_NAME:4240 for HTTP/2, and $NODE_NAME:4340 for gRPC.  The NODE_NAME
+# env var can be set from the downward API.
+#
+# There are sections of this config that can be uncommented to enable:
+# * CNI compatibility
+# * Automatic retries
+# * Zipkin tracing
+################################################################################
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: l5d-config
+data:
+  config.yaml: |-
+    admin:
+      port: 9990
+
+    namers:
+    # The io.l5d.k8s.{http, h2, grpc} namers are used when sending outgoing
+    # requests.  They look up the address of the target applicaiton and then
+    # send the request to the Linkerd DaemonSet pod running on the same node as
+    # the target application.
+    - kind: io.l5d.k8s
+      prefix: /io.l5d.k8s.http # We reference this in the http-outgoing router's dtab
+      transformers:
+      - kind: io.l5d.k8s.daemonset # This namer has the daemonset transformer "built-in"
+        namespace: default
+        port: http-incoming
+        service: l5d
+        # hostNetwork: true # Uncomment if using host networking (eg for CNI)
+    - kind: io.l5d.k8s
+      prefix: /io.l5d.k8s.h2 # We reference this in the h2-outgoing router's dtab
+      transformers:
+      - kind: io.l5d.k8s.daemonset # This namer has the daemonset transformer "built-in"
+        namespace: default
+        port: h2-incoming
+        service: l5d
+        # hostNetwork: true # Uncomment if using host networking (eg for CNI)
+    - kind: io.l5d.k8s
+      prefix: /io.l5d.k8s.grpc # We reference this in the grpc-outgoing router's dtab
+      transformers:
+      - kind: io.l5d.k8s.daemonset # This namer has the daemonset transformer "built-in"
+        namespace: default
+        port: grpc-incoming
+        service: l5d
+        # hostNetwork: true # Uncomment if using host networking (eg for CNI)
+
+    # The "basic" k8s namer.  We reference this in the incoming router's dtab
+    - kind: io.l5d.k8s
+
+    telemetry:
+    # Expose Prometheus style metrics on :9990/admin/metrics/prometheus
+    - kind: io.l5d.prometheus
+    # This can adversely affect performance at medium-high traffic volumes.
+    # The sample rate should be lowered to an acceptable level before going to
+    # production.
+    - kind: io.l5d.recentRequests
+      sampleRate: 0.25
+    # - kind: io.l5d.zipkin # Uncomment to enable exporting of zipkin traces
+    #   host: zipkin-collector.default.svc.cluster.local # Zipkin collector address
+    #   port: 9410
+    #   sampleRate: 1.0 # Set to a lower sample rate depending on your traffic volume
+
+    # This is used for anonymized usage reporting.  You can set the orgId to
+    # identify your organization or set `enabled: false` to disable entirely.
+    usage:
+      orgId: linkerd-examples-serviemesh
+
+    routers:
```

Comment:
Similarly, a section description would be nice here. "Here we configure what Linkerd actually does with incoming requests. We set up three `routers`, each listening on a distinct port, one for HTTP, one for HTTP/2 and one for gRPC. Etc..."